src/main/java/com/code_intelligence/jazzer/Jazzer.java

```diff
@@ -228,22 +233,26 @@ private static String prepareArgv0(Map<String, String> additionalEnvironment) th
                      .stream()
                      .map(e -> e.getKey() + "='" + e.getValue() + "'")
                      .collect(joining(" "));
-    String command = Stream
-                         .concat(Stream.of(javaBinary().toString()), javaBinaryArgs())
-                         // Escape individual arguments for the shell.
-                         .map(str -> shellQuote + str + shellQuote)
-                         .collect(joining(" "));
+    String command =
+        Stream
+            .concat(Stream.of(IS_ANDROID ? "exec" : javaBinary().toString()), javaBinaryArgs())
+            // Escape individual arguments for the shell.
+            .map(str -> shellQuote + str + shellQuote)
+            .collect(joining(" "));
 
     String invocation = env.isEmpty() ? command : env + " " + command;
-    String launcherContent = String.format(launcherTemplate, invocation);
 
     // argv0 is printed by libFuzzer during reproduction, so have the launcher basename contain
     // "jazzer".
     Path launcher;
+    String launcherContent;
     if (IS_ANDROID) {
+      String exportCommand = AndroidRuntime.getClassPathsCommand();
+      launcherContent = String.format(launcherTemplate, exportCommand, invocation);
```

Comment:
With this change "launcherTemplate" for android should be:

#!/system/bin/env sh\n%s\n%s $@\n

Reply:
Done. This was implemented previously but got changed when I rebased.